src/contract.rs

```diff
@@ -675,23 +809,46 @@ pub fn execute_pass_through_swap(
 
     validate_input_amount(&info.funds, input_token_amount, &input_token.denom)?;
 
+    let fees = FEES.load(deps.storage)?;
+    let total_fee_percent = fees.lp_fee_percent + fees.protocol_fee_percent;
     let amount_to_transfer = get_input_price(
         input_token_amount,
         input_token.reserve,
         transfer_token.reserve,
+        total_fee_percent,
     )?;
 
-    // Transfer tokens to contract
+    // Calculate fees
+    let protocol_fee_amount =
+        get_protocol_fee_amount(input_token_amount, fees.protocol_fee_percent)?;
+    let input_amount_with_protocol_fee = input_token_amount - protocol_fee_amount;
```

Comment:
again, nit, but feel `input_amount_minus_protocol_fee` or `input_amount_without_protocol_fee` is more clear